File: Python/Algorithms/selectionSort.py
'''
    @Author Jared Scott 
    This function will sort the given list of integers
'''

def main():
    list = [2,6,5,7,8,12,34,65,11,77,9,4,32]
    if type(list[0]) != type(1):
        print("Invalid list elements\nList must contain integers")
    else:    
        print("Before: " + str(list))
        selectionSort(list)
        print("After: " + str(list))   

def selectionSort(list):
    #This function sorts the list. 
    #NOTE: The original list has been modified
    for i in range(len(list)):
        minIndex = i
        for j in range(i+1,len(list)):
            if list[j] < list[minIndex]:
                minIndex = j
        list[i], list[minIndex] = list[minIndex],list[i]
    
main()

File: JavaScript/Algorithms/selection_sort.js
/*
 *@Author Jared Scott ☯
 *This script is an implementation of the selection sort algorithm for sorting integer arrays 
 */
 
function swap(array, idx1, idx2) {
	//Swaps the values at the given indices 
    var temp = array[idx1];
    array[idx1] = array[idx2];
    array[idx2] = temp;
}

function selectionSort(array, item_count) {
    var i, j, index;
    for (i = 0; i < item_count-1; i++)
    {
        //Scan through the array and find the minimum element 
        index = i;
        for (j = i + 1; j < item_count; j++) {
            if (array[j] < array[index])
                index = j;
		}
        //Once the smallest element has been found swap with current position
        swap(array,index, i);
    }
}

var array = [52,22,68,95,12,4,75,23,74];
selectionSort(array, array.length);
console.log("Sorted array: " + array);